Mathlib/LinearAlgebra/Dimension/Free.lean

```diff
@@ -165,6 +165,11 @@ variable {M M'}
 
 namespace Module
 
+/-- A free module of rank zero is trivial. -/
+lemma subsingleton_of_rank_zero (h : Module.rank R M = 0) : Subsingleton M := by
+  rw [← Basis.mk_eq_rank'' (Module.Free.chooseBasis R M), Cardinal.mk_eq_zero_iff] at h
+  exact (Module.Free.repr R M).subsingleton
```

Comment:
This should be added near `rank_zero_iff` instead.

Reply:
I thought that lemma would imply this one and I was wrong. I think the original place is better. Sorry for the back and forth.